src/Compilers/CSharp/Portable/Syntax/SyntaxFacts.cs

```diff
@@ -176,7 +176,16 @@ public static bool IsInTypeOnlyContext(ExpressionSyntax node)
                         return true;
 
                     case CrefParameter:
-                        return true;
+                        return ((CrefParameterSyntax)parent).Type == node;
+
+                    case TypeCref:
+                        // From the docs: "The syntax in a TypeCrefSyntax will always be bound as
+                        // type, so it's safer to use QualifiedCrefSyntax or MemberCrefSyntax if the
+                        // symbol might be a non-type member."
+                        //
+                        // As such QualifiedCrefSyntax and MemberCrefSyntax are in
+                        // IsInNamespaceOrTypeContext
+                        return ((TypeCrefSyntax)parent).Type == node;
```

Comment:
note: i'm not certain abou this.  The docs say it will always be bound as a type.  However, the actual code does this:

```c#
        private ImmutableArray<Symbol> BindTypeCref(TypeCrefSyntax syntax, out Symbol ambiguityWinner, DiagnosticBag diagnostics)
        {
            NamespaceOrTypeSymbol result = BindNamespaceOrTypeSymbolInCref(syntax.Type);
```

Which implies the compiler may see this as a namespace.  Thoughts?

Reply:
If the compiler impl is correct, then should likely update the docs to say 

```
// The syntax in a TypeCrefSyntax will always be bound as as **namespace** or type ...
```

I would also update this PR to move this check into IsInNamespaceOrTypeContext.

Reply:
> Which implies the compiler may see this as a namespace.  Thoughts? 

We should test what would doc comment compiler do if the type syntax refers to a namespace. Depending on that we keep this here or move to IsInNamespaceOrTypeContext 

---
In reply to: [360670924](https://github.com/dotnet/roslyn/pull/40549#discussion_r360670924) [](ancestors = 360670924)

Reply:
> We should test what would doc comment compiler do if the type syntax refers to a namespace. Depending on that we keep this here or move to IsInNamespaceOrTypeContext

Will do!  Thanks!